api/src/main/java/com/google/sps/SendNotifications.java

```
@@ -1,117 +1,155 @@
 
         // Adding player to Email object
-        emails.add(new Email(new User(playerEmail, playerName)));
-
-        // Adds Start game message to the email object
-        if(emails.isEmpty()) {
+        emails.add(emailType.createEmail(gameID, playerEmail, hostName));
+        
+        // Checks if there are any players
+        if (emails.isEmpty()) {
           throw new IllegalArgumentException();
-        } else {
-          emails.get(emails.size()-1).playerTurn(gameID);
         }
       }
     } catch (Exception e) {
-      System.out.println(e);
+      System.out.println("Exception in gatherRecipients" + e);
     }
 
     return emails;
   }
 
-  @GetMapping("/notifyTurn")
-  public void playerTurnEmail() throws IOException {
-    List<Email> emails = gatherRecipients();
+  public EmailCreation getNextPlayer(HttpServletRequest request, HttpServletResponse response)
+    throws IOException {
+
+    Firestore db = FirestoreClient.getFirestore();
+    String gameID = request.getParameter("gameID");
+    DocumentReference game = db.collection("games").document(gameID);
+    CollectionReference players = db.collection("users");
+    EmailCreation notification = null;
+
+    try {
+      DocumentSnapshot gameDocSnap = game.get().get();
+      int currentPlayerIndex = (int) ((long) gameDocSnap.get("currentPlayerIndex"));
+      String playerName = ((List<String>) gameDocSnap.get("players")).get(currentPlayerIndex);
+
+      DocumentSnapshot usersDocSnap = players.document(playerName).get().get();
+
+      // Retrieves player information
+      String playerEmail = usersDocSnap.getString("email");
+
+      notification = EmailCreation.playerTurnEmail(gameID, new User(playerEmail, playerName));
+    } catch (Exception e) {
+      System.out.println("Exception with DocumentSnapshot" + e);
+    }
+
+    if (notification == null) {
+      throw new IllegalArgumentException();
+    }
+
```

no need to do this.

---

Also, it is better to throw an Exception with a message describing the issue.